core/node/state_keeper/src/mempool_actor.rs

```diff
@@ -89,12 +89,30 @@ impl MempoolFetcher {
                 .await
                 .context("failed getting pending protocol version")?;
 
-            let l2_tx_filter = l2_tx_filter(
-                self.batch_fee_input_provider.as_ref(),
-                protocol_version.into(),
-            )
-            .await
-            .context("failed creating L2 transaction filter")?;
+            let l2_tx_filter = if let Some(unsealed_batch) = storage
+                .blocks_dal()
+                .get_unsealed_l1_batch()
+                .await
+                .context("failed getting unsealed batch")?
+            {
+                let (base_fee, gas_per_pubdata) = derive_base_fee_and_gas_per_pubdata(
+                    unsealed_batch.fee_input,
+                    protocol_version.into(),
+                );
+
+                L2TxFilter {
```

Comment:
Nit: creating `L2TxFilter` looks unnecessary here; IIUC, only `fee_per_gas` and `gas_per_pubdata` fields are used below, so it could be possible to return `derive_base_fee_and_gas_per_pubdata()` output directly.